File: Jason's Theorem.cpp
#include <bits/stdc++.h>
#define ll long long
#define ull unsigned long long
#define MOD 1000000007
#define vi vector<int>
#define pb push_back
#define pii pair<int, int>
#define mp make_pair
#define f first
#define s second
using namespace std;

ull N;

inline ull FPow (ull x, ull y) {
	if (y == 0) return 1;
	if (y == 1) return x;
	if (y & 1) return x * FPow((x * x) % MOD, y >> 1) % MOD;
	return FPow((x * x) % MOD, y >> 1) % MOD;
}

inline ull Calc (ull x) {
	return (ull) (FPow(2, x + 3) - 5 + MOD) % MOD;
}

int main () {
	scanf("%llu", &N);
	return 0 * printf("%llu\n", Calc(N));
}

File: Jason's Theorem.py
import sys
raw_input = sys.stdin.readline
N = 1L * int(raw_input())
MOD = 1000000007

def Fpow(x, y):
	if y == 0:
		return 1
	if y == 1:
		return x
	if y % 2 == 1:
		return x * Fpow((x * x) % MOD, y >> 1) % MOD
	else:
		return Fpow((x * x) % MOD, y >> 1) % MOD

def Calc(x):
	return 1L * (Fpow(2, N + 3) % MOD - 5 + MOD) % MOD

print((Calc(N) + MOD) % MOD)